### ISSUE TITLE
DataTable: Missing rows when filtering with VirtualScroller in Firefox

### ISSUE THREAD

**Issue by sasa0103** [primefaces/primereact#5118]
### Describe the bug

When you scroll down some rows and use the filter, no result is shown. In Chrome it works, but not in Firefox. I used the example from the documentation (virtualscroller) and added some filters to reproduce the problem in my application.

### Reproducer

https://codesandbox.io/s/primereact-demo-forked-kfn7nh?file=/src/App.js:745-754

### PrimeReact version

10.0.5

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

Firefox 118.0.2 (64-bit)

### Steps to reproduce the behavior

1. Open example with table and filters
2. Copy "Vin" value from first row
3. Scroll down to e.g. line 20
4. Enter copied text to filter
5. Entry is not shown

### Expected behavior

Row with the copied entry should be shown.

**Comment by sasa0103** [comment]
Any idea when this will be fixed?

**Comment by melloware** [comment]
No idea unless you have a PrimeTek PRO account and can request this fix through your PRO support.

**Comment by heshamwhite** [comment]
> There is a temporary workaround: manually invoke the `scrollInView` when the items changed

On which element should scrollInView be invoked? 
Because I tried to call it on the p-datatable,p-datatable-table  and p-datatable-tbody without success. 
It couldn't be invoked on a "tr" as the tbody is completely empty hence "Missing rows"

**Comment by kl-nevermore** [comment]
```tsx
const tableRef = useRef()
<DataTable
        ref={tableRef}
        value={cars}
        filters={filters}
        filterDisplay="row"
        globalFilterFields={["vin", "year", "brand", "color"]}
        scrollable
        scrollHeight="400px"
        virtualScrollerOptions={{ itemSize: 46 }}
        tableStyle={{ minWidth: "50rem" }}
        onValueChange={(e) => {
        // try it
          const vs = tableRef.current.getVirtualScroller();
          tableRef.scrollInView();
        }}
      >
```

**Comment by heshamwhite** [comment]
Thank you for your reply but neither the tableRef object not the tableRef.current has a scrollIntoView function 

> tableRef.scrollInView is not a function

> TypeError: tableRef.current.scrollInView is not a function

am I missing something?

[Here](https://codesandbox.io/p/sandbox/primereact-demo-forked-kfn7nh?file=%2Fsrc%2FApp.js%3A23%2C26) is a sandbox of the change.

**Comment by kl-nevermore** [comment]
I pasted it wrong


```tsx
 onValueChange={(e) => {
        // try it
          const vs = tableRef.current.getVirtualScroller();
          // here
          vs.scrollInView();
        }}
```

**Comment by kl-nevermore** [comment]
![20240307161630_rec_](https://github.com/primefaces/primereact/assets/25167742/7c2feb41-2db4-4ae1-8f00-344892bb5413)

**Comment by heshamwhite** [comment]
super, yes it works. Thank you for your help!

Hopefully the original bug would be fixed soon

**Comment by ssode** [comment]
Hey, I recently came across the same issue and tried applying the workaround mentioned here, but noticed that when using TypeScript `VirtualScroller.scrollInView` expects at least 2 arguments. It seems to work for the filtering issue either way if casted, but now I'm seeing some weird behavior if I scroll down a bit and then sort on a column. The scroll bar will jump to a seemingly random position and not display any rows until the user manually scrolls the table again whereas in Chrome doing the same thing will reset the scroll position to the top and rows will be displayed normally.

https://stackblitz.com/edit/k45amg?file=src%2FApp.tsx

**Comment by melloware** [comment]
I don't see `scrollIntoView` in your reproducer?

**Comment by ssode** [comment]
There is no `scrollIntoView` method on VirtualScroller unless I'm missing something here? I call `scrollInView` on line line 43, but as I mentioned I cannot even call it in the same manner that the workaround uses without typecasting it.

**Comment by melloware** [comment]
what is this statement then? `"VirtualScroller.scrollInView expects at least 2 arguments"`

**Comment by melloware** [comment]
OH I see the typescript needs to say props are optional on scrollInView

**Comment by ssode** [comment]
Exactly yeah, are you also able to see the other issue I mentioned with sorting in Firefox?

**Comment by melloware** [comment]
I didn't check that other virtual scroller in firefox issue but i believe you.

**Comment by melloware** [comment]
@ssode PR submitted to fix the typescript but I didn't make Index optional i made the `to` value optional.    Can you review?

**Comment by ssode** [comment]
Hey @melloware, looks good. I think what was happening when calling `scrollInView` with no params was it would fall through to the last case where it calls `scrollToIndex` and in there there seems to be some logic that would default the index to 0.

**Comment by melloware** [comment]
yep it got closed when i merged my PR I reopened it as I only fixed the Typescript issue.
